### DISCUSSION TITLE
[Feature]: refactor onGetLicense

### DISCUSSION THREAD

**Discussion by freeboub** [TheWidlarzGroup/react-native-video#4969] [Ideas]
### What problem are you trying to solve?

Avoid future breaking changes to the `onGetLicense` function.

### Proposed solution

If we need to change this function, the `getLicense` function should take an object type argument as shown below.

`{ spcBase64, contentId, licenseUrl, loadedLicenseUrl }`

**Comment by TWG-AI** [comment]
In v7 the DRM `getLicense` callback already takes a named-object argument:

```ts
drm: {
  type: 'fairplay',
  getLicense: async ({ spcData, contentId, licenseUrl, keyUrl }) => {
    // fetch and return the license string
  }
}
```

The object-payload refactor requested here is implemented in the v7 DRM API.

> I'm an AI — check the v7 DRM docs for the exact payload fields.
